### DISCUSSION TITLE
Is predictorMatrix doing right?

### DISCUSSION THREAD

**Discussion by iagogv3** [amices/mice#361] [Q&A]
I'm not sure how to understand the use of `predictorMatrix`. After reading in https://stefvanbuuren.name/fimd/sec-modelform.html that in
```
> set.seed(1); imp <- mice(nhanes, print = FALSE)
> imp$predictorMatrix
    age bmi hyp chl
age   0   1   1   1
bmi   1   0   1   1
hyp   1   1   0   1
chl   1   1   1   0
```

> The rows correspond to incomplete target variables, in the sequence as they appear in the data. A value of `1` indicates that the column variable is a predictor to impute the target (row) variable, and a `0` means that it is not used. Thus, in the above example, `bmi` is predicted from `age`, `hyp` and `chl`. Note that the diagonal is `0` since a variable cannot predict itself. Since `age` contains no missing data, `mice()` silently sets all values in the row to `0`. The default setting of the `predictorMatrix` specifies that every variable predicts all others.

Then I would understand that:

1. first, as for `age`, I would understand that if I set all values of `bmi` row to 0, then `bmi` missings wouldn't be predicted. But it isn't the case.
2. further, if those row values only affect to the prediction of `bmi`, shouldn't the predictions of the other variables be equal (under `set.seed(1)`)? (unless predictions for the other variables depend on the predictions made for `bmi`?)

Can you help me?

Thank you!

**Comment by stefvanbuuren** [comment]
Thanks for your interest. 

I am afraid I do not understand what you do not understand. Perhaps running some examples might help to increase your level of comfort. If that results in questions that I can understand, I'd be perfectly willing to answer.

Hope this helps, Stef.

**Comment by stefvanbuuren** [comment]
Note: "mice() silently sets all values in the row to 0" is old behaviour, and not true anymore.
